x-pack/plugins/rollup/public/search/rollup_search_strategy.js

```diff
@@ -95,15 +95,17 @@ export const rollupSearchStrategy = {
       failedSearchRequests,
     } = await serializeAllFetchParams(allFetchParams, searchRequests);
 
-    const abortable = kfetch({
+    const controller = new AbortController();
+    const promise = kfetch({
+      signal: controller.signal,
       pathname: '../api/rollup/search',
       method: 'POST',
       body: serializedFetchParams,
     });
 
     return {
       searching: new Promise((resolve, reject) => {
```

Comment:
Would you mind updating this to not use the `Promise` constructor? It just creates an unnecessary separation in the promise chain.

If, at some point in the future, the error comes back **without** a `body` property then the `.catch()` handler would throw, but never makes its way to the `reject()` function, and the `searching` promise would just never resolve. Ideally that should never happen.